cmd/ndm_daemonset/probe/mountprobe_test.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package probe
+
+import (
+	"os"
+	"os/exec"
+	"path"
+	"syscall"
+	"testing"
+
+	"github.com/openebs/node-disk-manager/pkg/mount"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/openebs/node-disk-manager/blockdevice"
+	"github.com/openebs/node-disk-manager/pkg/udev"
+)
+
+const mountsFilePath string = "/proc/1/mounts"
+
+func TestMountProbeFillBlockDeviceDetails(t *testing.T) {
+	mockDisk, err := udev.MockDiskDetails()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	mp := &mountProbe{}
+
+	t.Run("DevPath empty", func(t *testing.T) {
+		bd := blockdevice.BlockDevice{}
+		mp.FillBlockDeviceDetails(&bd)
+		if len(bd.FSInfo.MountPoint) > 0 {
+			t.Fail()
+		}
+	})
+
+	t.Run("Device not mounted", func(t *testing.T) {
+		// Chroot into a tmp dir
+		err := enterFakeRoot(t)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer exitFakeRoot()
+
+		bd := blockdevice.BlockDevice{
+			Identifier: blockdevice.Identifier{
+				DevPath: rand.String(2),
+			},
+		}
+		mp.FillBlockDeviceDetails(&bd)
+		if len(bd.FSInfo.MountPoint) > 0 {
+			t.Fail()
+		}
+	})
+
+	t.Run("Device mounted", func(t *testing.T) {
+		err := enterFakeRoot(t)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer exitFakeRoot()
+
+		bd := blockdevice.BlockDevice{
+			Identifier: blockdevice.Identifier{
+				DevPath: mockDisk.Dependents.Partitions[0],
+			},
+		}
+		mp.FillBlockDeviceDetails(&bd)
+		if len(bd.FSInfo.MountPoint) <= 0 {
```

Comment:
Can we compare the actual mountpoint instead of just checking if a mountpoint is present or not.?